pkg/oc/admin/router/router.go

```diff
@@ -683,6 +683,9 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out, errout io.Write
 	}
 	// automatically start the internal metrics agent if we are handling a known type
 	if cfg.Type == "haproxy-router" {
+		if cfg.StatsPort == 0 {
+			return fmt.Errorf("When --type is haproxy-router, --stats-port must not be 0")
```

Comment:
This is incorrect, type has nothing to do with stats port.

Reply:
@smarterclayton This is your code:
db5b0fbc pkg/cmd/admin/router/router.go        (Clayton Coleman         2017-03-10 01:43:10 -0500 685)  if cfg.Type == "haproxy-router" {
6e9c0ada pkg/oc/admin/router/router.go         (Phil Cameron            2017-09-29 15:39:40 -0400 686)          if cfg.StatsPort == 0 {
6e9c0ada pkg/oc/admin/router/router.go         (Phil Cameron            2017-09-29 15:39:40 -0400 687)                  return fmt.Errorf("When --typ
e is haproxy-router, --stats-port must not be 0")
6e9c0ada pkg/oc/admin/router/router.go         (Phil Cameron            2017-09-29 15:39:40 -0400 688)          }
ff44d237 pkg/cmd/admin/router/router.go        (Clayton Coleman         2017-06-20 21:38:36 -0400 689)          env["ROUTER_LISTEN_ADDR"] = fmt.Sprin
tf("0.0.0.0:%d", cfg.StatsPort)
db5b0fbc pkg/cmd/admin/router/router.go        (Clayton Coleman         2017-03-10 01:43:10 -0500 690)          env["ROUTER_METRICS_TYPE"] = "haproxy
"
db5b0fbc pkg/cmd/admin/router/router.go        (Clayton Coleman         2017-03-10 01:43:10 -0500 691)  }

You use cfg.StatsPort to set up the  ROUTER_LISTEN_ADDR - 0.0.0.0:0 is the error that causes the router to not come up.

Reply:
Sorry, wrong package.  If stats-port is 0 AND type is haproxy-router, then in this code path you should avoid setting ROUTER_METRICS_TYPE and listen addr, not return this error.  This is defaulting metrics on.

```
if cfg.Type == "haproxy-router" && cfg.StatsPort != 0 {
  env["ROUTER_LISTEN_ADDR"] = fmt.Sprintf("0.0.0.0:%d", cfg.StatsPort)
  env["ROUTER_METRICS_TYPE"] = "haproxy"
}
```